docs/helper-environment-visitor.md

```diff
@@ -0,0 +1,285 @@
+---
+id: babel-helper-environment-visitor
+title: "@babel/helper-environment-visitor"
+sidebar_label: helper-environment-visitor
+---
+
+`@babel/helper-environment-visitor` is a utility package that provides a current `this` context visitor.
+
+## Installation
+
+```shell npm2yarn
+npm install @babel/helper-environment-visitor
+```
+
+## Usage
+
+To use the package in your Babel plugin, import the required functions from `@babel/helper-environment-visitor`:
+
+```js title="my-babel-plugin.js"
+import environmentVisitor, {
+  requeueComputedKeyAndDecorators
+} from "@babel/helper-environment-visitor";
+```
+
+### `environmentVisitor`
+
+It visits all AST nodes within the same `this` context to the root traverse node. Running this visitor alone is no-op as it does not modify AST nodes. This visitor is meant to be used with `traverse.visitors.merge`.
+
+```js title="collect-await-expression.plugin.js"
+module.exports = (api) => {
+  const { types: t, traverse } = api;
+  return {
+    name: "collect-await",
+    visitor: {
+      Function(path) {
+        if (path.node.async) {
+          const awaitExpressions = [];
+          // Get a list of related await expressions within the async function body
+          path.traverse(traverse.visitors.merge([
+            environmentVisitor,
+            {
+              AwaitExpression(path) {
+                awaitExpressions.push(path);
+              },
+              ArrowFunctionExpression(path) {
+                path.skip();
+              },
+            }
+          ]))
+        }
+      }
+    }
+  }
+}
+```
+
+### `requeueComputedKeyAndDecorators`
+
+```typescript
+requeueComputedKeyAndDecorators(path: NodePath): void
+```
+
+Requeue the computed key and decorators of a class member `path` so that they will be revisited after current traversal queue is drain. See the [example](#replace-top-level-this) section for more usage.
```

Comment:
```suggestion
Requeue the computed key and decorators of a class member `path` so that they will be revisited after current traversal queue is drained. See the [example](#replace-top-level-this) section for more usage.
```